clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1742,8 +1742,15 @@ private void maybeUpdateAssignment(SubscriptionState subscription) {
                     if (!this.assignedPartitions.contains(tp)) {
                         MetricName metricName = partitionPreferredReadReplicaMetricName(tp);
                         if (metrics.metric(metricName) == null) {
-                            metrics.addMetric(metricName, (Gauge<Integer>) (config, now) ->
-                                subscription.preferredReadReplica(tp, 0L).orElse(-1));
+                            metrics.addMetric(metricName, (Gauge<Integer>) (config, now) -> {
+                                try {
+                                    return subscription.preferredReadReplica(tp, 0L).orElse(-1);
+                                } catch (IllegalStateException e) {
```

Comment:
Perhaps it's reasonable in this case to let `preferredReadReplica` use `assignedStateOrNull`?